File: code/mathematical-algorithms/Check_is_square/check_is_square.swift
#!/usr/bin/env swift
// Part of Cosmos by OpenGenus Foundation
import Darwin

func findingSquare(num : Int) -> Bool {
    let x = Int(sqrt(Double(num)))
    for i in 0...x{
        if(i*i == num){
            return true
        }
    }
    return false
}
let number = Int(readLine()!)
let result = findingSquare(num: number!)

if(result){
    print("A perfect square")
}
else{
    print("Not a perfect square")
}

File: Check_is_Square.java
public class Check_is_Square {

	static boolean checkIsSquare(int number)
	{
		double sqrt = Math.sqrt(number);
		int x = (int) sqrt;
		if(Math.pow(sqrt,2) == Math.pow(x,2))
		{
			return true;
		}
		else 
		{
			return false;
		}
	}
	
	public static void main(String[] args) {
		int a = 9;
		int b = 10;
		
		System.out.println("Is " + a + " a square number? " + checkIsSquare(a));  //true
		System.out.println("Is " + b + " a square number? " + checkIsSquare(b));  //false
	}

}
